### PR TITLE
Refresh clip_grad patch.

### PR DESCRIPTION
**Pull request by ailzhang**
For easier review, here's the real diff:
```     
if norm_type == inf:
-        total_norm = max(p.grad.detach().abs().max().to(device) for p in parameters)
+        total_norm = max(p.grad.detach().abs().max() for p in parameters)
     else:
-        total_norm = torch.norm(torch.stack([torch.norm(p.grad.detach(), norm_type).to(device) for p in parameters]), norm_type)
-    clip_coef = max_norm / (total_norm + 1e-6)
-    if clip_coef < 1:
-        for p in parameters:
-            p.grad.detach().mul_(clip_coef.to(p.grad.device))
+        total_norm = torch.norm(torch.stack([torch.norm(p.grad.detach(), norm_type) for p in parameters]), norm_type)
+    clip_coef = torch.tensor(max_norm, device=device) / (total_norm + 1e-6)
+    clip_value = torch.where(clip_coef < 1, clip_coef, torch.tensor(1., device=device))
+    for p in parameters:
+        p.grad.detach().mul_(clip_value)
```

For now XLA would not support cross device gradient clipping as we did in https://github.com/pytorch/pytorch/pull/38615. We could re-evaluate whether to allow this API support across device later.

### CODE REVIEW

**Comment by dlibenzi** [comment]
Does this need to go in @ailzhang ?

**Comment by ailzhang** [comment]
The Pytorch PR is landing soon, let's merge it after that ;)
